OneSignalSDK/unittest/src/test/java/com/test/onesignal/InAppMessagingUnitTests.java

```diff
@@ -616,4 +616,19 @@ public void testOnMessageWasShown() throws Exception {
         assertEquals(1, iamImpressionRequest.payload.get("device_type"));
         assertEquals(true, iamImpressionRequest.payload.get("first_impression"));
     }
+
+    @Test
+    public void testOnPageChanged() throws Exception {
+        threadAndTaskWait();
```

Comment:
is this needed?

Reply:
yes it crashes without it

Reply:
oh, weird 🤔  test setup should not break tests, can you point out which error it is?

Reply:
Ah sorry the crash can be fixed, but you need to initialize to have the player id etc for the request